ibis/backends/__init__.py

```diff
@@ -1269,6 +1270,108 @@ def has_operation(cls, operation: type[ops.Value]) -> bool:
             f"{cls.name} backend has not implemented `has_operation` API"
         )
 
+    @util.experimental
+    def read_csv(
+        self, path: str | Path, table_name: str | None = None, **kwargs: Any
+    ) -> ir.Table:
+        """Register a CSV file as a table in the current backend.
+
+        This function reads a CSV file and registers it as a table in the current
+        backend. Note that for Impala and Trino backends, CSV read performance
+        may be suboptimal.
+
+        Parameters
+        ----------
+        path
+            The data source. A string or Path to the CSV file.
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+        **kwargs
+            Additional keyword arguments passed to the backend loading function.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+
+        Examples
+        --------
+        Connect to a SQLite database:
+
+        >>> con = ibis.sqlite.connect()
+
+        Read a single csv file:
+
+        >>> table = con.read_csv("path/to/file.csv")
+
+        Read all csv files in a directory:
+
+        >>> table = con.read_parquet("path/to/csv_directory/*")
+
+        Read all csv files with a glob pattern:
+
+        >>> table = con.read_csv("path/to/csv_directory/test_*.csv")
+
+        Read csv file from s3:
+
+        >>> table = con.read_csv("s3://bucket/path/to/file.csv")
+
+        """
+        pa = self._import_pyarrow()
+        import pyarrow.csv as pcsv
+        from pyarrow import fs
+
+        read_options_args = {}
+        parse_options_args = {}
+        convert_options_args = {}
+        memory_pool = None
+
+        for key, value in kwargs.items():
+            if hasattr(pcsv.ReadOptions, key):
+                read_options_args[key] = value
+            elif hasattr(pcsv.ParseOptions, key):
+                parse_options_args[key] = value
+            elif hasattr(pcsv.ConvertOptions, key):
+                convert_options_args[key] = value
+            elif key == "memory_pool":
+                memory_pool = value
+            else:
+                raise ValueError(f"Invalid args: {key!r}")
```

Comment:
There should be a link in the docstring, at least, to indicate where users can look for what valid options are.

And if a user passes in their own dictionary of `ReadOptions`, should we accept that?

Reply:
We might also want to add examples in the docstrings (and tests) of common CSV arguments, like `skip_rows` and `column_names`

Reply:
> And if a user passes in their own dictionary of `ReadOptions`, should we accept that?

Great. Thanks for your comments. I could add support for doing this. 

I have some followups, taking the readOptions as an example, it could be
1. dictionary of ReadOptions , i.e. 
```python
read_options_dict = {'skip_rows': 1, 'column_names': ['animals', 'n_legs', 'entry']
con.read_csv(path/to/csv, read_options=read_options_dict)
```
2. ReadOptions Object,  i.e.
```python
read_options = csv.ReadOptions(autogenerate_column_names=True, skip_rows=1)
con.read_csv(path/to/csv, read_options=read_options)
```
3. attributes of ReadOptions:
```python
con.read_csv(path/to/csv, skip_rows=1, autogenerate_column_names=True)
```
Should we support all these cases?
Do we need to consider if there is a conflicts between 2 and 3, for example:
```python
read_options = csv.ReadOptions(autogenerate_column_names=True, skip_rows=1)
# skip_rows is defined twice
con.read_csv(path/to/csv, read_options=read_options, skip_rows=2)
```

Thanks again for your input.

Reply:
I like to think about how I would describe the available options in a single sentence.  Once that sentence gets too complicated, we're doing too much.

I think we could support options 2 & 3, and collisions are handled by overriding the contents of e.g. `ReadOptions` with the explicit kwarg.  So in your example above, we would skip the first 2 rows.

Reply:
Argh, you know, I think I just changed my mind.  If anyone needs hyper-specific setup for reading a csv with pyarrow, they can just read in the CSV with `pyarrow` themselves and then pass in the resulting table.

So ignore the above.

Reply:
I _do_ think that having to look at 3 different pages in the pyarrow docs for all the options is _very_ annoying, so we can explicitly document common options in the `kwargs` section of the docstring and then point to the various upstream pages for users who want a full list.